### PR TITLE
Add Authorization tests to Spar regarding Scim User management

### PR DESCRIPTION
**Pull request by ChrisPenner**
Purpose: Ensure that user management is secure and auth is respected

### CODE REVIEW

File: services/spar/test-integration/Test/Spar/Scim/UserSpec.hs [hunk 1]
```diff
@@ -83,6 +85,39 @@ testExternalIdIsRequired = do
...
+    env <- ask
+
+    user1 <- randomScimUser
+    user2 <- randomScimUser
+    user3 <- randomScimUser
+    (tokTeamA, _) <- registerIdPAndScimToken
+    (tokTeamB, _) <- registerIdPAndScimToken
+
+    -- Create and add a first user: success!
+    createUser_ (Just tokTeamA) user1 (env ^. teSpar)
```
Comment (neongreen) [inline services/spar/test-integration/Test/Spar/Scim/UserSpec.hs]: You can use `createUser` here.

**Comment by neongreen** [comment]
Oops, merged and only then realized it wasn't my branch :/

**Comment by neongreen** [comment]
I hope it's fine with this one, but if it wasn't ready then yell.

**Comment by fisx** [comment]
it was double-approved, so sure.  :-)

**Pull request merged**